src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -170,7 +171,8 @@ public function reverseTransform($value)
         }
 
         try {
-            $dateTime = new \DateTime(sprintf(
+            $dateTimeClass = $this->getDateTimeClass();
+            $dateTime = new $dateTimeClass(sprintf(
```

Comment:
`@return \DateTime` line 111 should be changed to `@return \DataTime|\DateTimeImmutable` and also in other transformers